src/qforte/bindings.cc

```diff
@@ -20,13 +21,21 @@ namespace py = pybind11;
 using namespace pybind11::literals;
 
 PYBIND11_MODULE(qforte, m) {
-    py::class_<Circuit>(m, "Circuit")
+    py::class_<Circuit, std::shared_ptr<Circuit>>(m, "Circuit")
         .def(py::init<>())
+        .def(py::init<const Circuit&>())
         .def("add", &Circuit::add_gate)
         .def("add", &Circuit::add_circuit)
         .def("add_gate", &Circuit::add_gate)
         .def("add_circuit", &Circuit::add_circuit)
+        .def("insert_gate", &Circuit::insert_gate)
+        .def("remove_gate", &Circuit::remove_gate)
+        .def("swap_gates", &Circuit::swap_gates)
+        .def("insert_circuit", &Circuit::insert_circuit)
+        .def("remove_gates", &Circuit::remove_gates)
```

Comment:
Please either include a docstring specifying that this removes gates in a _range_ of positions or rename the function to `remove_gate_interval` or some such. The name makes sense if I already know what the function does, but what the function does isn't obvious if all I know is the name and the function signature.

Reply:
Thanks for this suggestion.